docs/add-ins/add-and-remove-attachments-to-an-item-in-a-compose-form.md

```diff
@@ -101,33 +111,76 @@ function addItemAttachment(itemId) {
+You can use the [getAttachmentsAsync](/office/dev/add-ins/reference/objectmodel/preview-requirement-set/office.context.mailbox.item#getattachmentsasyncoptions-callback--arrayattachmentdetails) method to get the attachments of the message or appointment being composed.
 
-The following JavaScript function, `removeAttachment`, continues to extend the examples above, and removes the specified attachment from the email or appointment that is being composed. The function takes as an argument the ID of the attachment to be removed. You can obtain the ID of an attachment after a successful `addFileAttachmentAsync` or `addItemAttachmentAsync` method call, and store it for a subsequent `removeAttachmentAsync` method call.
+To get an attachment's content, you can use the [getAttachmentContentAsync](/office/dev/add-ins/reference/objectmodel/preview-requirement-set/office.context.mailbox.item#getattachmentcontentasyncattachmentid-options-callback--attachmentcontent) method. The supported formats are listed in the [AttachmentContentFormat](/javascript/api/outlook/office.mailboxenums.attachmentcontentformat) enum.
+
+You should provide a callback method to check for the status and any error by using the `AsyncResult` output parameter object. You can also pass any additional parameters to the callback method by using the optional `asyncContext` parameter.
+
+The following JavaScript example gets the attachments and allows you to set up distinct handling for each supported attachment format.
+
+```js
+var item = Office.context.mailbox.item;
+var options = {asyncContext: {currentItem: item}};
+item.getAttachmentsAsync(options, callback);
+
+function callback(result) {
+  if (result.value.length > 0) {
+    for (i = 0 ; i < result.value.length ; i++) {
+      result.asyncContext.currentItem.getAttachmentContentAsync(result.value[i].id, handleAttachmentsCallback);
+    }
+  }
+}
+
+function handleAttachmentsCallback(result) {
+  // Parse string to be a url, an .eml file, a base64-encoded string, or an .icalendar file.
+  if (result.value.format === Office.MailboxEnums.AttachmentContentFormat.Base64) {
```

Comment:
Not a big deal, but consider making this a switch statement. That's usually easier to read than a series of `else if`s.